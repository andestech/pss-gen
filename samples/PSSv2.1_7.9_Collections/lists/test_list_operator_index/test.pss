/*!
@weakgroup      test
@{
@file
@author         Luther Lee
@data           2023/10/25
@brief          Test list index operator `[]`.
@version
Portable Test and Stimulus Standard Version 2.1 <br>
> Section 7.9 : Collections <br>
> Section 7.9.3 : Lists <br>


---------------------------------------------------------------------------------------------------

### Test Command
@code{.unparsed}
make gentarget
@endcode
*/
component pss_top {

    enum eSTR2NUM {
        ZERO, ONE, TWO
    };

    struct sSTR2NUM {
        string stringVal = "NULL";
        int intVal = 0;
    };

    action root_a {
        list<bit [4] > nibbleList = {4'b0010, 4'b0011};
        list<int     > intList    = {2, 3};
        list<bool    > boolList   ;
        list<eSTR2NUM> enumList   ;
        list<string  > stringList = {"A", "B"};
        list<float32 > float32List;
        list<float64 > float64List;
        // list<chandle > chandleList;
        list<sSTR2NUM> structList ;

        bit [4] nibbleVal;
        int     intVal   ;
        string  stringVal;
        exec post_solve {
            nibbleVal = nibbleList[0];
            intVal    = intList[0];
            stringVal = stringList[0];
        }

        exec body ASM =
"""
// test_list_operator_index
nibbleVal = {{nibbleVal}}
intVal    = {{intVal}}
stringVal = {{stringVal}}
""";
    }
}

/**@}*/
