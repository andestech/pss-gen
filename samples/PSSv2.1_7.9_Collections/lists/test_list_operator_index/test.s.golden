
// test_list_operator_index
nibbleVal = 2
intVal    = 2
stringVal = A

